ic-asset/src/asset_config.rs

```diff
@@ -0,0 +1,627 @@
+        // 2. order of rules withing file matters - last rule in config file takes precedens over the first one
+        let cfg = Some(HashMap::from([
+            (
+                "".to_string(),
+                r#"[
+        {"match": "nested/**/*", "cache": {"max_age": 900}},
+        {"match": "nested/deep/*", "cache": {"max_age": 800}},
+        {"match": "nested/**/*.toml","cache": {"max_age": 700}}
+    ]"#
+                .to_string(),
+            ),
+            (
+                "nested".to_string(),
+                r#"[
+        {"match": "the-thing.txt", "cache": {"max_age": 600}},
+        {"match": "*.txt", "cache": {"max_age": 500}},
+        {"match": "*", "cache": {"max_age": 400}}
+    ]"#
+                .to_string(),
+            ),
+            (
+                "nested/deep".to_string(),
+                r#"[
+        {"match": "**/*", "cache": {"max_age": 300}},
+        {"match": "*", "cache": {"max_age": 200}},
+        {"match": "*.toml", "cache": {"max_age": 100}}
+    ]"#
+                .to_string(),
+            ),
+        ]));
+        let assets_temp_dir = create_temporary_assets_directory(cfg, 7).unwrap();
+        let assets_dir = assets_temp_dir.path();
+
+        println!("ff");
+        let assets_config = dbg!(AssetSourceDirectoryConfiguration::load(assets_dir))?;
+        for f in [
+            "index.html",
+            "js/index.js",
+            "js/index.map.js",
+            "css/main.css",
+            "css/stylish.css",
+        ] {
+            assert_eq!(
+                assets_config.get_asset_config(assets_dir.join(f).as_path()),
+                AssetConfig::default()
+            );
+        }
+
+        assert_eq!(
+            assets_config.get_asset_config(assets_dir.join("nested/the-thing.txt").as_path()),
+            AssetConfig {
+                cache: Some(CacheConfig { max_age: 400 }),
+                ..Default::default()
+            },
+        );
+        assert_eq!(
+            assets_config
+                .get_asset_config(assets_dir.join("nested/deep/the-next-thing.toml").as_path()),
+            AssetConfig {
+                cache: Some(CacheConfig { max_age: 100 }),
+                ..Default::default()
+            },
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    #[should_panic]
+    fn no_content_config_file() {
+        let cfg = Some(HashMap::from([
+            ("".to_string(), "".to_string()),
+            ("css".to_string(), "".to_string()),
+            ("js".to_string(), "".to_string()),
+            ("nested".to_string(), "".to_string()),
+            ("nested/deep".to_string(), "".to_string()),
+        ]));
+        let assets_temp_dir = create_temporary_assets_directory(cfg, 0).unwrap();
+        let assets_dir = assets_temp_dir.path();
+        let assets_config = AssetSourceDirectoryConfiguration::load(assets_dir);
+        assets_config.unwrap();
```

Comment:
Let's check for errors rather than panics, because `should_panic` would pass if the panic happened anywhere in the test, whereas we want the final `assets_config` to be an error

Reply:
done in https://github.com/dfinity/agent-rs/pull/352/commits/6cfe6a556eedc9d6684c97d0f7d9d0d3461787f0